### PR TITLE
feat: picture selection question

### LINKED ISSUE #1256: [FEATURE][FormTribe 🔥][750 Points] Question Type: Picture Choice
### Welcome to the FormTribe

This issue will get you the points mentioned in issue title for the [FormTribe Hackathon](https://formtribe.com)

**Preparation:**
If this is your first time contributing, please read:

- Our [introductory blog post](https://formbricks.com/blog/join-the-formtribe)
- Overview of [Code Guidelines at Formbricks.](https://formbricks.com/docs/contributing/how-we-code)

**Only when your PR follows _all_ of the best practices it will get merged.**

Want more points quick? Complete [Side Quests 🏰 ](https://formbricks.com/formtribe#side-quests)

--------

### Is your feature request related to a problem? Please describe.

Users want to let their respondents pick 1 of several images

### Describe the solution you'd like

Offer the corresponding question type:

image

Note: This mockup is stemming from our upcoming UI revamp. The UI you will build should match the current look of our surveys 🤓

### Tasks

1. Create a new question type called "Picture Selection" with HeroIcon "Photo"
image

2. Create the corresponding question settings card with the following settings. We do have a reusable component for the file upload UI being used in the Welcome Card:

image
image

The Allow Multi Select toggle switches the type from a Radio button functionality (pick one) to a Checkbox functionality (pick many). **Please have a look at the question types Single and Multi Select, you will be able to adopt most of the logic from there.**

3. Create a UI to display the results in the Survey Summary: 
image

4. Create a UI to display the results in the Response Card: Its pretty straightforward, just do it consistent with the others responses 😃 

5. Test extensively in:
- Survey editor preview
- Link Surveys
- Web app surveys

### PR DESCRIPTION
**Pull request by gupta-piyush19**
## What does this PR do?
Added Picture selection question

Fixes #1256 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change adds a new database migration
- [ ] This change requires a documentation update

## How should this be tested?

- Test A
- Test B

### Required

- [ ] Filled out the "How to test" section in this PR
- [ ] Read [How we Code at Formbricks](<[https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md](https://formbricks.com/docs/contributing/how-we-code)>)
- [ ] Self-reviewed my own code
- [ ] Commented on my code in hard-to-understand bits
- [ ] Ran `pnpm build`
- [ ] Checked for warnings, there are none
- [ ] Removed all `console.logs`
- [ ] Merged the latest changes from main onto my branch with `git pull origin main`
- [ ] My changes don't cause any responsiveness issues
- [ ] First PR at Formbricks? [Please sign the CLA!](https://formbricks.com/clmyhzfrymr4ko00hycsg1tvx) Without it we wont be able to merge it 🙏

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

**Comment by jobenjada** [comment]
Hey @gupta-piyush19 - thanks for shipping this! :)

Here is my feedback:

**1. Add a senseful default**

Pick 2 cute puppies from unsplash, compress them down and add them to the public folder or so so we can display the same 2 pictures in every account.

Add "Which is the cutest puppie?" as question placeholder.

--------

**2. Remove scroll bar**
Please remove the sroll bar on the left hand side:

image

---------

**3. Don't skew images**

Use display type fill so that the images don't get skewed

image

-------

**4. When you select two images and then toggle multi select off, you get this:**

image

Instead, the selection should be removed when the toggle is activated.

-------

**5. Reduce text size to xs**

image

--------

**6. Dont display it in black / white when not selected**

I'm aware that this was in the mock ups but it skews the image too much. E.g. when the question is "Which color do you like more?" or for things like "Which logo do you like more?" people need to be able to compare both images colored.

image

--------

**7. Add space pls**

image

-------

**8. Add description**

All other question types have a description, we should allow people to add one here as well :)

image

-------

9. Display images in response card ;) 

image

---------

### Thanks!!!

File: packages/ui/FileInput/index.tsx
Comment (pandeymangg) [inline packages/ui/FileInput/index.tsx:209]: please use a different array for handing the allowed file types for preview, you could make an array like
`allowedFileTypesForPreview` and use a `includes` on it to do this, also, please include the extension `webp` in the `allowedFileExtensions` type as well as here 🙂

**Review by pandeymangg — COMMENTED** [review]
Thanks a lot for the PR @gupta-piyush19 ! Looks and works great 🥳 
Just left a few comments, after that we can merge this 😊

**Comment by jobenjada** [comment]
from my side, it's good to go :) @pandeymangg merge when you approve my changes

**Review by pandeymangg — APPROVED** [review]
LGTM!

**Pull request merged**
